camel-core/src/main/java/org/apache/camel/processor/Throttler.java

```diff
@@ -192,6 +217,26 @@ public boolean process(final Exchange exchange, final AsyncCallback callback) {
         }
     }
 
+    private DelayQueue<ThrottlePermit> locateDelayQueue(final Integer key) throws InterruptedException, ExecutionException {        
+        CompletableFuture<DelayQueue<ThrottlePermit>> futureDelayQueue = new CompletableFuture<>();
+
+        delayQueueCacheExecutorService.submit(() -> {
+            futureDelayQueue.complete(findDelayQueue(key));
+        });
+        DelayQueue<ThrottlePermit> currentQueue = futureDelayQueue.get();   
```

Comment:
futrueDelayQueue.get() is wait for the future method to complete, it doesn't save us time.

Reply:
My intention was not save time, was to fill up the throttle group async

Reply:
I agree with Willem, I don't see yet the value of doing this async.

Reply:
maybe not very big plus. and 
i disagree. In case of async routing, could be useful.